src/Symfony/Component/Validator/Constraints/NotSuspiciousValidator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Contracts\Translation\LocaleAwareInterface;
+
+/**
+ * @author Mathieu Lechat <mathieu.lechat@les-tilleuls.coop>
+ */
+class NotSuspiciousValidator extends ConstraintValidator implements LocaleAwareInterface
+{
+    private string $locale;
+
+    public function validate(mixed $value, Constraint $constraint)
+    {
+        if (!$constraint instanceof NotSuspicious) {
+            throw new UnexpectedTypeException($constraint, NotSuspicious::class);
+        }
+
+        if (null === $value || '' === $value) {
```

Comment:
Your checking two times '' === $value

Reply:
And the Is_scalar return false for null so I think you can remove this line

Reply:
> Your checking two times '' === $value

The second check comes after the casting of `$value`, which could yield an empty string.

> is_scalar return false for null so I think you can remove this line

Well we don’t want to throw an exception when validating `null`!
See the rationale [here](https://symfony.com/doc/current/validation/custom_constraint.html#creating-the-validator-itself):

> constraints should ignore null and empty values to allow other constraints (NotBlank, NotNull, etc.) to take care of that

FYI these lines come from the `EmailValidator`